ginkgo_linter.go

```diff
@@ -371,6 +368,19 @@ func checkAsyncAssertion(pass *analysis.Pass, config types.Config, expr *ast.Cal
 	return true
 }
 
+func isValidFuncCallInAsync(t gotypes.Type) bool {
+	switch t.(type) {
+	// allow functions that return function or channel.
+	case *gotypes.Signature, *gotypes.Chan, *gotypes.Pointer:
```

Comment:
Nit: I find the intention easier to read when the `return true` is here instead of below.